ecs/cluster.yaml

```diff
@@ -909,22 +936,38 @@ Resources:
       AutoScalingGroupName: !Ref AutoScalingGroup
       Cooldown: 300
       ScalingAdjustment: -25
-  CPUReservationHighAlarm:
+  ContainerInstancesShortageAlarm:
     Type: 'AWS::CloudWatch::Alarm'
     Properties:
-      AlarmDescription: 'Cluster is running out of CPU (reservation)'
-      Namespace: 'AWS/ECS'
+      AlarmDescription: 'Cluster is running out of container instances'
+      Namespace: !Ref 'AWS::StackName'
       Dimensions:
       - Name: ClusterName
         Value: !Ref Cluster
-      MetricName: CPUReservation
-      ComparisonOperator: GreaterThanThreshold
-      Statistic: Average # special rule because we scale on reservations and not utilization
+      MetricName: SchedulableContainers
+      ComparisonOperator: LessThanOrEqualToThreshold
+      Statistic: Minimum # special rule because we scale on reservations and not utilization
       Period: 60
       EvaluationPeriods: 1
-      Threshold: 80
+      Threshold: !Ref ContainerShortageThreshold
       AlarmActions:
       - !Ref ScaleUpPolicy
+  ContainerInstancesExcessAlarm:
+    Type: 'AWS::CloudWatch::Alarm'
+    Properties:
+      AlarmDescription: 'Cluster is wasting container instances'
+      Namespace: !Ref 'AWS::StackName'
+      Dimensions:
+      - Name: ClusterName
+        Value: !Ref Cluster
+      MetricName: SchedulableContainers
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      Statistic: Maximum # special rule because we scale on reservations and not utilization
+      Period: 60
+      EvaluationPeriods: 1
+      Threshold: !Ref ContainerExcessThreshold
+      AlarmActions:
+      - !Ref ScaleDownPolicy
   CPUReservationTooHighAlarm:
```

Comment:
If containers are added faster that auto scaling can scale this metric can help?